src/lib/datepicker/year-view.ts

```diff
@@ -95,6 +95,24 @@ export class MdYearView implements AfterContentInit {
 
   /** Creates an MdCalendarCell for the given month. */
   private _createCellForMonth(month: number) {
-    return new MdCalendarCell(month, this._locale.shortMonths[month].toLocaleUpperCase());
+    return new MdCalendarCell(
+        month, this._locale.shortMonths[month].toLocaleUpperCase(), this._isMonthEnabled(month));
+  }
+
+  /** Whether the given month is enabled. */
+  private _isMonthEnabled(month: number) {
+    if (!this.dateFilter) {
+      return true;
+    }
+
+    let enabled = false;
+    for (let date = new SimpleDate(this.date.year, month, 1); date.month === month;
+         date = date.add({days: 1})) {
+      enabled = enabled || this.dateFilter(date);
```

Comment:
Do you need the `enabled ||` check? Looks like if it's enabled, it will break anyway.